src/components/events/event_new/EventNewForm.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Form from 'react-bootstrap/Form';
+import { Button } from 'react-bootstrap';
+import './EventNewForm.css';
+import { ToggleUseDates } from './fields/ToggleUseDates';
+import { DateField } from './fields/DateField';
+import { LocationField } from './fields/LocationField';
+import { CostField } from './fields/CostField';
+import { DescriptionField } from './fields/DescriptionField';
+
+export class EventNewForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      useDates: true,
+      eventType: props.eventType,
+      ...props.event
```

Comment:
I think I see what's going on here... It's not quite following React's standard of having only one point of truth for a piece of data. That is, the parent has a state with the event, it passes the state as a property, and then we copy the state, update the local state (but not the parent's state), format that state, and pass it back (throwing away the parent's old state). I think it's probably fine to do `...props.event` here, but I question why you save the getState() in the parent's state (couldn't you directly pass it in without having to save it as the state of the parent?)

Reply:
That's a good point. I suppose I could just inject both methods into EventNewForm and then work with them there. I'll make those changes.